base_newgen_payment_card/models/newgen_payment_card.py

```diff
@@ -23,6 +23,9 @@ class NewgenPaymentCard(models.Model):
         'account.journal', string='Bank Journal', check_company=True,
         domain="[('type', '=', 'bank'), ('company_id', '=', company_id)]",
         ondelete='restrict')
+    purchase_journal_id = fields.Many2one(
+        'account.journal', string='Purchase journal', ondelete='restrict',
+        domain="[('type', '=', 'purchase')]")
```

Comment:
On the definition of the field purchase_journal_id, could you add check_company=True and add ('company_id', '=', company_id) in the domain. You can take inspiration from the definition of the field journal_id

Reply:
ok, it's done